streams/shared/src/main/scala/zio/stream/ZSink.scala

```diff
@@ -1446,15 +1308,15 @@ object ZSink extends ZSinkPlatformSpecific {
    * Creates a sink by starts consuming value as soon as one fails
    * the predicate `p`.
    */
-  final def ignoreWhile[A](p: A => Boolean): ZSink[Any, Nothing, A, A, Unit] =
+  final def ignoreWhile[A](p: A => Boolean): ZSink[Any, Nothing, A, Unit] =
```

Comment:
I'd like this to be a `SinkPure` and not delegate through `ignoreWhileM`. Ignore if you think it's not worth it.

Reply:
Yeah we'll do that in a follow-up. There are several combinators that can be overriden in `SinkPure`.